### ISSUE TITLE
bisync: retries and delay on locked files

### ISSUE THREAD

**Issue by martind69** [rclone/rclone#7555]
In relation to #7471 i will split off this issue as an extra feature request.

I tried using --retries=1000 --retries-sleep=30 for cases where MS Office files are occupied or locked!
The bisync task will end up with a fatal error and needs to be resynced.

@nielash [posted a preview.](https://github.com/nielash/rclone/releases/tag/v1.65.0-beta.7566.68eb6f5b2.bisync-feature-preview-2024-01-01) **Many THANKS for that!**
I share my test results with this preview beta here:

```
rclone bisync "path1" "C:\TBD\Test" --config="rclone.conf" --workdir="cache" --exclude="{~$*,*.tmp,*.bak}" --check-first --fast-list --remove-empty-dirs --retries=1000 --retries-sleep=30 --conflict-resolve=newer --conflict-loser=delete --resilient --recover --max-lock 5m -v
2024/01/09 06:31:11 NOTICE: bisync is IN BETA. Don't use in production!
2024/01/09 06:31:11 INFO  : Setting --ignore-listing-checksum as neither --checksum nor --compare checksum are set.
2024/01/09 06:31:11 INFO  : Bisyncing with Comparison Settings:
{
        "Modtime": true,
        "Size": true,
        "Checksum": false,
        "HashType1": 0,
        "HashType2": 0,
        "NoSlowHash": false,
        "SlowHashSyncOnly": false,
        "SlowHashDetected": true,
        "DownloadHash": false
}
2024/01/09 06:31:11 INFO  : lock file renewed for 5m0s. New expiration: 2024-01-09 06:36:11.3875714 +0100 CET m=+300.422610101
2024/01/09 06:31:11 INFO  : Synching Path1 "path1" with Path2 "C:\TBD\Test\"
2024/01/09 06:31:11 INFO  : Building Path1 and Path2 listings
2024/01/09 06:31:11 INFO  : Path1 checking for diffs
2024/01/09 06:31:11 INFO  : - Path1    File changed: size (smaller), time (newer) - Ship-to form.docx
2024/01/09 06:31:11 INFO  : Path1:    1 changes:    0 new,    1 modified,    0 deleted
2024/01/09 06:31:11 INFO  : (Modified:    1 newer,    0 older,    0 larger,    1 smaller)
2024/01/09 06:31:11 INFO  : Path2 checking for diffs
2024/01/09 06:31:11 INFO  : Applying changes
2024/01/09 06:31:11 INFO  : - Path1    Queue copy to Path2       - C:\TBD\Test\Ship-to form.docx
2024/01/09 06:31:11 INFO  : - Path1    Do queued copies to                - Path2
2024/01/09 06:31:11 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/09 06:31:11 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/09 06:31:11 ERROR : Ship-to form.docx.cagiyuw2.partial: partial file rename failed: rename \\?\C:\TBD\Test\Ship-to form.docx.cagiyuw2.partial \\?\C:\TBD\Test\Ship-to form.docx: Access is denied.
2024/01/09 06:31:11 INFO  : Ship-to form.docx.cagiyuw2.partial: Removing failed copy
2024/01/09 06:31:11 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/09 06:31:11 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/09 06:31:11 NOTICE: copy1to2: Received error: rename \\?\C:\TBD\Test\Ship-to form.docx.cagiyuw2.partial \\?\C:\TBD\Test\Ship-to form.docx: Access is denied. - retrying as --resilient is set. Retry 1/1000
2024/01/09 06:31:11 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/09 06:31:11 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/09 06:31:11 ERROR : Ship-to form.docx.zeluram5.partial: partial file rename failed: rename \\?\C:\TBD\Test\Ship-to form.docx.zeluram5.partial \\?\C:\TBD\Test\Ship-to form.docx: Access is denied.
2024/01/09 06:31:11 INFO  : Ship-to form.docx.zeluram5.partial: Removing failed copy
2024/01/09 06:31:11 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/09 06:31:11 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/09 06:31:11 NOTICE: copy1to2: Received error: rename \\?\C:\TBD\Test\Ship-to form.docx.zeluram5.partial \\?\C:\TBD\Test\Ship-to form.docx: Access is denied. - retrying as --resilient is set. Retry 2/1000
```

This loop runs very quick without any sleep time. So the `--retries-sleep=30` was ignored!

At this moment i unlocked the file.

```
2024/01/09 06:31:26 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/09 06:31:26 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/09 06:31:26 INFO  : Ship-to form.docx: Copied (replaced existing)
2024/01/09 06:31:26 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/09 06:31:26 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/09 06:31:26 NOTICE: copy1to2: Received error: not deleting files as there were IO errors - retrying as --resilient is set. Retry 174/1000
2024/01/09 06:31:26 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/09 06:31:26 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/09 06:31:26 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/09 06:31:26 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/09 06:31:26 NOTICE: copy1to2: Received error: not deleting files as there were IO errors - retrying as --resilient is set. Retry 175/1000
```

The loop keeps running...
So i ended up with sending an exit signal.

```
2024/01/09 06:31:31 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/09 06:31:31 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/09 06:31:31 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/09 06:31:31 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: not deleting files as there were IO errors - retrying as --resilient is set. Retry 281/1000
2024/01/09 06:31:31 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/09 06:31:31 INFO  : Signal received: interrupt
2024/01/09 06:31:31 NOTICE: Attempting to gracefully shutdown. (Send exit signal again for immediate un-graceful shutdown.)
2024/01/09 06:31:31 INFO  : Telling Sync to wrap up early.
2024/01/09 06:31:31 INFO  : Canceling Sync if not done in: 30
2024/01/09 06:31:31 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/09 06:31:31 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/09 06:31:31 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: not deleting files as there were IO errors - retrying as --resilient is set. Retry 282/1000
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 283/1000
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 284/1000
...
...
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 999/1000
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 1000/1000
2024/01/09 06:31:31 INFO  : Updating listings
2024/01/09 06:31:31 INFO  : Validating listings for Path1 "path1" vs Path2 "C:\TBD\Test\"
2024/01/09 06:31:31 INFO  : Removing empty directories
2024/01/09 06:31:31 NOTICE: Graceful shutdown completed successfully.
2024/01/09 06:31:31 INFO  : Bisync successful
2024/01/09 06:31:31 INFO  :
Transferred:       14.388 MiB / 14.388 MiB, 100%, 711.230 KiB/s, ETA 0s
Errors:               173 (retrying may help)
Checks:               396 / 396, 100%
Transferred:            1 / 1, 100%
Elapsed time:        20.5s

2024/01/09 06:31:31 Failed to bisync with 173 errors: last error was: rename \\?\C:\TBD\Test\Ship-to form.docx.lebeqic0.partial \\?\C:\TBD\Test\Ship-to form.docx: Access is denied.
2024/01/09 06:31:32 INFO  : Exiting...
```

I was wondering why 14.388 MiB are transferred. Every loop the partial file was copied.
So the `--retries-sleep=30` would be reduce the amount.

My summary of the bisync-feature-preview:
anything causes tons of verbose output.
stability seems good. next run recovers automatically.

**Comment by nielash** [comment]
Thank you, @martind69! This is great feedback. I think there are two issues here, both with easy fixes:

- Bisync should reset the error count in between each retry (it currently does not -- this is why it kept retrying after you unlocked the file.)
- Bisync should implement support for `--retries-sleep`

I should be able to address these pretty quickly (and post a new version to try, if you'd like!)

> anything causes tons of verbose output.

It is verbose because the `-v` flag was included -- you can omit this flag to make it less verbose (or add `-q` for even less). That said, if there are certain messages you think should be downgraded from `NOTICE` to `INFO` or `INFO` to `DEBUG`, let me know and I'm happy to consider changing them 🙂 

> stability seems good. next run recovers automatically.

Glad to hear it! 🙂

**Comment by martind69** [comment]
I'm glad to hear from you!

> I should be able to address these pretty quickly (and post a new version to try, if you'd like!)

That were great! I'm your debugger 🙂

> It is verbose because the `-v` flag was included

I know that 🙂, what i mean is this:

```
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: not deleting files as there were IO errors - retrying as --resilient is set. Retry 282/1000
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 283/1000
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 284/1000
...
...
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 999/1000
2024/01/09 06:31:31 NOTICE: copy1to2: Received error: <nil> - retrying as --resilient is set. Retry 1000/1000
```
why is that? the copy was successfull before

`2024/01/09 06:31:26 INFO  : Ship-to form.docx: Copied (replaced existing)`

And btw, the unit isn't print out.

`2024/01/09 06:31:31 INFO  : Canceling Sync if not done in: 30`

**Comment by nielash** [comment]
> That were great! I'm your debugger 🙂

I appreciate that! [Here's a new version](https://github.com/nielash/rclone/releases/tag/v1.65.0-beta.7569.5bd90ea42.fix-7555-bisync-retries) which I think should fix these issues.

> why is that? the copy was successfull before

Silly oversight on my part -- I neglected to exit the loop early if `err` came back `nil`. The verbosity was more of a symptom -- it should not have gotten past retry 282 in the first place! 🙂 It should be fixed in this version.

> And btw, the unit isn't print out.

This is actually a 30-second countdown that logs every second during the graceful shutdown sequence -- but yours finished in less than a second so you never got to 29 😄 But I agree it would be clearer with a unit -- I've added it in this new version. Thanks for spotting that 🙂 

Let me know how the new version goes for you if you have a chance to try it!

**Comment by martind69** [comment]
Good morning! in relation to "goog" i have test results ;)

```
...
2024/01/10 08:59:14 INFO  : Backup listing is valid.
2024/01/10 08:59:14 INFO  : Building Path1 and Path2 listings
2024/01/10 08:59:14 INFO  : Path1 checking for diffs
2024/01/10 08:59:14 INFO  : - Path1    File changed: size (larger), time (newer) - Ship-to form.docx
2024/01/10 08:59:14 INFO  : Path1:    1 changes:    0 new,    1 modified,    0 deleted
2024/01/10 08:59:14 INFO  : (Modified:    1 newer,    0 older,    1 larger,    0 smaller)
2024/01/10 08:59:14 INFO  : Path2 checking for diffs
2024/01/10 08:59:14 INFO  : Applying changes
2024/01/10 08:59:14 INFO  : - Path1    Queue copy to Path2       - C:\TBD\Test\Ship-to form.docx
2024/01/10 08:59:14 INFO  : - Path1    Do queued copies to                - Path2
2024/01/10 08:59:14 INFO  : Local file system at //?/C:/TBD/Test: Running all checks before starting transfers
2024/01/10 08:59:14 INFO  : Ship-to form.docx: Deleted
2024/01/10 08:59:14 INFO  : Ship-to form.docx: Copied (replaced existing)
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 1/10 sleeping 1ms
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 2/10 sleeping 2ms
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 3/10 sleeping 4ms
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 4/10 sleeping 8ms
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 5/10 sleeping 16ms
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 6/10 sleeping 32ms
2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 7/10 sleeping 64ms
2024/01/10 08:59:15 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 8/10 sleeping 128ms
2024/01/10 08:59:15 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 9/10 sleeping 256ms
2024/01/10 08:59:15 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 10/10 sleeping 512ms
2024/01/10 08:59:15 ERROR : Ship-to form.docx: Couldn't delete: remove \\?\C:\TBD\Test\Ship-to form.docx: The process cannot access the file because it is being used by another process.
2024/01/10 08:59:15 INFO  : Local file system at //?/C:/TBD/Test: Checks finished, now starting transfers
2024/01/10 08:59:15 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors
2024/01/10 08:59:15 ERROR : Local file system at //?/C:/TBD/Test: not deleting directories as there were IO errors
2024/01/10 08:59:15 NOTICE: copy1to2: Received error: remove \\?\C:\TBD\Test\Ship-to form.docx: The process cannot access the file because it is being used by another process. - retrying as --resilient is set. Retry 1/1000
2024/01/10 08:59:15 NOTICE: will retry in 10s at 2024-01-10 08:59:25 CET
```

Question: what is the purpose of the 2^0 ... 2^9 [ms] stepping?
`2024/01/10 08:59:14 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 1/10 sleeping 1ms`

Bug: without verbosity we end up in an critical error
```
2024/01/10 08:58:48 NOTICE: \\?\C:\TBD\Test\Ship-to form.docx: Remove detected sharing violation - retry 10/10 sleeping 512ms
2024/01/10 08:58:49 ERROR : Ship-to form.docx: Couldn't delete: remove \\?\C:\TBD\Test\Ship-to form.docx: The process cannot access the file because it is being used by another process.
2024/01/10 08:58:49 ERROR : Ship-to form.docx: Couldn't move into backup dir: remove \\?\C:\TBD\Test\Ship-to form.docx: The process cannot access the file because it is being used by another process.
2024/01/10 08:58:49 ERROR : Bisync critical error: C:\TBD\Test\ delete failed for C:\TBD\Test\Ship-to form.docx: remove \\?\C:\TBD\Test\Ship-to form.docx: The process cannot access the file because it is being used by another process.
2024/01/10 08:58:49 ERROR : Bisync aborted. Must run --resync to recover.
```

Ugly: if i choose error log to `--log-file` it would log tons of ERROR (in case without -v but that's currently not possible). That aren't real errors in my case, it needs a reclassification.

Feature idea: i thinked about using `--suffix` in addition to `--backup-dirN`
But there is no dynamic option yet. I think about `--suffix datetime` like microtime or `--suffix date` only date.
And a really nice option would be a `--backup-cleanup` with max-age definition.
What do you think about that?

**Comment by nielash** [comment]
> Question: what is the purpose of the 2^0 ... 2^9 [ms] stepping?

These are ["low-level-retries"](https://rclone.org/docs/#low-level-retries-number), which are different from the "high-level retries" controlled by the `--retries` flag. They are built in to rclone and not specific to bisync. Basically the idea is to retry certain low-level errors in case they are just temporary, and wait for longer and longer amounts between each retry.

In your case, the `Remove detected sharing violation` error seems to be related to a Windows-specific quirk -- there is some context here: https://github.com/rclone/rclone/commit/42f0963bf907daacb6cb718ef14a64a591fab3a0 

> Bug: without verbosity we end up in an critical error

I'm not sure if I understand this comment. Do you mean that the behavior is different with and without the `-v` flag? Or is the issue just that rclone is erroring for something you don't consider an error (regardless of the `-v` flag)?

> Ugly: if i choose error log to `--log-file` it would log tons of ERROR 

Not sure I understand... what do you wish would happen instead when there is an ERROR?

> That aren't real errors in my case, it needs a reclassification.

Can you give me more info about the use case and why it isn't a real error? It looks like you are trying to overwrite a `docx` file while it is open for editing on the destination... what do you wish would happen instead in that scenario? Is the idea that rclone should silently skip this file instead of complaining about it?

A few things that would be helpful to test:

- is the behavior different in `sync` vs. `bisync`?
- does using `--ignore-errors` help?
- does using `--low-level-retries 1` help?
- Can you give me instructions to reproduce this scenario? (I do have a Windows machine I can try)

> Feature idea: i thinked about using `--suffix` in addition to `--backup-dirN`
But there is no dynamic option yet. I think about `--suffix datetime` like microtime or `--suffix date` only date.

I agree this would be nice! I actually just developed a nearly identical feature for bisync's `--conflict-suffix` flag -- you can read about the dynamic date variables [here](https://github.com/nielash/rclone/blob/bisync-feature-preview-2024-01-01/docs/content/bisync.md#--conflict-suffix-stringstring-conflict-suffix). Use is like this:

```
--conflict-suffix {DateOnly}-conflict
// result: myfile.txt.2006-01-02-conflict1
```


The main [`--suffix`](https://rclone.org/docs/#suffix-suffix) flag is not specific to bisync, but I think it would be quite easy to move that same code into `operations` and use it for both, if people like that idea 🙂 

> And a really nice option would be a `--backup-cleanup` with max-age definition.
What do you think about that?

Explain a little more about what this would do? Would it be like running `rclone delete` on the backup-dir with a `--min-age` filter? Would it apply to all files or only the ones related to the current sync?

**Comment by martind69** [comment]
> > Bug: without verbosity we end up in an critical error
> 
> I'm not sure if I understand this comment. Do you mean that the behavior is different with and without the `-v` flag? Or is the issue just that rclone is erroring for something you don't consider an error (regardless of the `-v` flag)?

I mean that the behavior is different with and without the `-v` flag
In other words, it can be used only in verbose mode!

> > Ugly: if i choose error log to `--log-file` it would log tons of ERROR
> 
> Not sure I understand... what do you wish would happen instead when there is an ERROR?

When i set up `--log-level ERROR` then i expect nothing in relation to my tests. But i see many error logs in verbose mode. Maybe it's necessary to do a reclassification because there are no real (top-level) errors. I think that are sub-level errors. Some examples:
`2024/01/10 08:59:15 ERROR : Ship-to form.docx: Couldn't delete: remove \\?\C:\TBD\Test\Ship-to form.docx: The process cannot access the file because it is being used by another process.`
Is not an error but maybe a warning?
`2024/01/10 08:59:15 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors`
Is not an error, it's a result of the above one.

> Is the idea that rclone should silently skip this file instead of complaining about it?

Yes! that's a great idea. Then you can report (log) a notice or info classification.

> A few things that would be helpful to test:
> 
> * is the behavior different in `sync` vs. `bisync`?
> * does using `--ignore-errors` help?
> * does using `--low-level-retries 1` help?
> * Can you give me instructions to reproduce this scenario? (I do have a Windows machine I can try)

I will test it soon, sorry no time today.

> I actually just developed a nearly identical feature for bisync's `--conflict-suffix` flag

Perfect, good idea!

> The main [`--suffix`](https://rclone.org/docs/#suffix-suffix) flag is not specific to bisync

I think you can implement this feature to bisync. My idea using `--suffix datetime` like microtime or `--suffix date` only date. For scenarios that no data loss can be accepted, this flag renames the files unique before move to `--backup-dirN`

> Explain a little more about what this would do? Would it be like running `rclone delete` on the backup-dir with a `--min-age` filter? Would it apply to all files or only the ones related to the current sync?

Yes that's correct! The idea is seen global - not to a specific sync. purge all where the `--min-age` filter is applicable.
Then users can plan with a storage budget and doesn't purge manually.

**Comment by nielash** [comment]
> I mean that the behavior is different with and without the `-v` flag
> In other words, it can be used only in verbose mode!

Can you send a complete log with `-v` and also without `-v`, so that I can see what you mean? (Your last `-v` log cuts off at `NOTICE: will retry in 10s at 2024-01-10 08:59:25 CET`, so I can't see what happened after that... did the retries succeed or fail?) 

So far, I have not seen anything unexpected in my tests, or in the excerpts you posted from your logs. Your OS failed to copy the file, so rclone logged this as an error and exited with an error status. That is what I'd expect to happen.

> When i set up `--log-level ERROR` then i expect nothing in relation to my tests. But i see many error logs in verbose mode.

`--log-level ERROR` still reports errors -- it just suppresses everything else. It sounds to me like the log-level is more of a side-effect, and the real issue here is that rclone is considering something to be an error that you do not consider to be an error. I think what you are saying is that you are not concerned about this `Couldn't delete` error, because bisync can try again on a future run, and sooner or later it will succeed, once the file is closed. Is that right? That sounds reasonable to me. What is tricky though is: how should rclone decide which errors are concerning to the user, and which errors are not? From rclone's perspective, its job was to copy a file, and it failed to do so. Rclone's expectation is that the dest should always match the source at the end of a sync, or else something went wrong.

I think probably the best solution would be a new flag to allow these kinds of errors to be ignored. But we still have to give rclone some criteria to determine which errors to care about. Do you have suggestions for how it should decide? (What is an example of an error you _would_ want to be fatal?)

> `2024/01/10 08:59:15 ERROR : Local file system at //?/C:/TBD/Test: not deleting files as there were IO errors`
> Is not an error, it's a result of the above one.

This one I am pretty sure can be ignored with the `--ignore-errors` flag. Or rather, the error will still be reported, but it will not be considered fatal and will not prevent the deletions. Let me know if you have a chance to try it.

> My idea using `--suffix datetime` like microtime or `--suffix date` only date. For scenarios that no data loss can be accepted, this flag renames the files unique before move to `--backup-dirN`

It's a great idea, and I think it would be useful for `sync` and `move` as well. `--conflict-suffix` already supports all of [these formats](https://pkg.go.dev/time#pkg-constants).

> Yes that's correct! The idea is seen global - not to a specific sync. purge all where the `--min-age` filter is applicable.
> Then users can plan with a storage budget and doesn't purge manually.

Makes sense. I think if it's global and not specific to a sync, then probably it makes more sense to just schedule a new cron job for `rclone delete [some:dir] --min-age [x]`. I'm not sure I see an advantage to triggering this from `sync` or `bisync`, and I think it could be problematic for a sync to be deleting things from a backup-dir that weren't related to the sync. But that's just my opinion -- feel free to open a new issue for it if you like, and it can be discussed!
